akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Demux.scala

```diff
@@ -289,7 +290,7 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
 
         pingState.tickInterval().foreach(interval =>
           // to limit overhead rather than constantly rescheduling a timer and looking at system time we use a constant timer
-          schedulePeriodically(ConfigurablePing.Tick, interval)
+          scheduleWithFixedDelay(ConfigurablePing.Tick, interval, interval)
```

Comment:
Note that `schedulePeriodically` is the same as `scheduleAtFixedRate` so this could change the rate of the ticks slightly. I don't know if that would matter here but in general `scheduleWithFixedDelay` is more safe since it doesn't generate bursts after pauses.

```
   * Warning: `scheduleAtFixedRate` can result in bursts of scheduled tasks after long
   * garbage collection pauses, which may in worst case cause undesired load on the system.
   * Therefore `scheduleWithFixedDelay` is often preferred.
```

Reply:
This one is used to tick itself, and check if it should keep a HTTP2 ping or has waited for too long for a pong and should timeout the connection, so worst case with a delayed tick is a connection was kept alive longer than the timeout before it actually timed out. I think that is better than bursts here.